src/test/java/htsjdk/variant/variantcontext/GenotypeLikelihoodsUnitTest.java

```diff
@@ -125,16 +125,36 @@ public void testGetAsMap(){
     }
 
     @Test
-    public void testCalculateNumLikelihoods() {    
-        
+    public void testCalculateNumLikelihoods() {
+
         for (int nAlleles=2; nAlleles<=5; nAlleles++)
-            // simplest case: diploid
+        {
             Assert.assertEquals(GenotypeLikelihoods.numLikelihoods(nAlleles, 2), nAlleles*(nAlleles+1)/2);
+        }
 
         // some special cases: ploidy = 20, #alleles = 4
         Assert.assertEquals(GenotypeLikelihoods.numLikelihoods(4, 20), 1771);
+        // ploidy = 10, alleles = 5
+        Assert.assertEquals(GenotypeLikelihoods.numLikelihoods(5, 10), 1001);
+        // ploidy = 16, alleles = 10
+        Assert.assertEquals(GenotypeLikelihoods.numLikelihoods(10, 16), 2042975);
+    }
+
+    @Test
+    public void testNumLikelihoodsCache() {
+        GenotypeNumLikelihoodsCache cache = new GenotypeNumLikelihoodsCache();
+
+        cache.put(2, 5, 10);
+        cache.put(10, 16, 2042975);
```

Comment:
since there's a bit of slightly tricky multilayered map access stuff going on, could we add an additional test case where we put a value that shares the same first level.  i.e. add `(2, 10, whateverthevalueis)` . In general we tend to use dataproviders for cases like this where we have a bunch of similar tests with different values, but this is fine if you don't want to change it over.

Reply:
I'll switch to a data provider.  And I think switching to the single-layered map will remove the potential of issues that could result from the multilayered one.  I'll probably make a few changes to these tests to account for the other changes